cub/cub/detail/choose_offset.cuh

```diff
@@ -56,6 +57,18 @@ struct ChooseOffsetT
                                          unsigned long long>::type;
 };
 
+/**
+ * common_iterator_value sets member type to the common_type of 
+ * value_type for all argument types. used to get OffsetT in 
+ * DeviceSegmentedReduce.
+ */
+template <typename... Iter>
+struct common_iterator_value {
+    using type = cuda::std::__common_type_t<cuda::std::__iter_value_type<Iter>...>;
```

Comment:
crucial: CUB is occasionally used in nested cuda namespaces (`thrust::system::cuda`), so in order to avoid ambiguity we have to specify the full name:
```suggestion
    using type = ::cuda::std::__common_type_t<::cuda::std::__iter_value_type<Iter>...>;
```